contents/docs/deployment/securing-posthog.md

```diff
@@ -31,6 +31,11 @@ TRUST_ALL_PROXIES=True
 
 > If you're on Heroku, you are behind a proxy by default, so you'll need to add `IS_BEHIND_PROXY=True`. Heroku automatically overrides `X-Forwarded-For`, so you can use `TRUST_ALL_PROXIES=True`.
 
+## Secure Cookies
+
+Starting with PostHog 1.13.0, we introduced a `SECURE_COOKIES` flag. This defaults to "False" when PostHog is running on `localhost` and "True" in production.
+
+**For security reasons, you should always have this set to "True" in production.** However, in some cases, such as while testing your setup, you may need to toggle this flag off. In that case, you can explicitly set `SECURE_COOKIES=False` as an environment variable. But always remember, live data should never be transferred if this flag is off!
```

Comment:
Hi, this explanation is not technical enough. It doesn't answer what the flag does nor why it exists. It just tries to scare the user into setting it to `True` (what is actually not needed) without explaining what's at stake.

Somebody, somewhere will need the exact technical details of what is happening and why... and will need to dig into our codebase to get the answers. 

To clarify the technical points:

> This defaults to "False" when PostHog is running on `localhost`

--> Not true... and what is "running on localhost"? Every machine is its own localhost. I get what was meant here (running on your own computer), but in reality this flag defaults to False when running PostHog in development or test mode (the flags `DEBUG` or `TEST` are set to `1` or `True`)

> For security reasons, you should always have this set to "True" in production.

--> You do not need to set this to `True` in production, it's like that already. There's no point to override a default - more hassle with the setup. However you need to make sure that you are **never** running `DEBUG=1` in production (which sets this flag to `False` as well). Otherwise there are worse things that can happen than slightly less secure cookies.

> ... such as while testing your setup, you may need to toggle this flag off.

--> As someone deploying posthog to my infrastructure, I really want to know in which part of "testing my setup" exactly will this flag come in handy. 

The short version is that:
- Setting this flag toggles the [`Secure`](https://en.wikipedia.org/wiki/Secure_cookie) flag on the Django Session, CSRF and Toolbar Login cookies.
- If you have posthog behind HTTPS (like every production setup should)... and this flag is False, Chrome will throw warnings about cookies and you might have trouble logging in in the next versions of Chrome. The toolbar login cookie won't work and you will also be vulnerable to MITM attacks when you accidentally open your app using "http" and not "https". 
- If you have posthog behind HTTP (you're setting up the app and haven't configured a HTTPS proxy yet) and this flag is True, you won't be able to log in to the app since secure cookies are discarded in an unsafe environment.

Hope this helps make the docs better :)

Reply:
Ok yeah these are great points! Will update